pkg/cluster/operation/destroy.go

```diff
@@ -237,10 +237,20 @@ func DestroyMonitored(getter ExecutorGetter, inst spec.Instance, options *spec.M
 func CleanupComponent(getter ExecutorGetter, instances []spec.Instance, cls spec.Topology, options Options) error {
 	retainDataRoles := set.NewStringSet(options.RetainDataRoles...)
 	retainDataNodes := set.NewStringSet(options.RetainDataNodes...)
+	cleanDataNodes := set.NewStringSet(options.CleanDataNodes...)
 
 	for _, ins := range instances {
+		dataRetained := false
```

Comment:
Could you please move those lines into a small function, such as: 

```go
checkInstanceRetained(instance spec.Instance, options Options) bool 
```

then we'll add more unit tests on this function.